model/src/main/kotlin/DependencyGraph.kt

```diff
@@ -326,17 +326,7 @@ class DependencyReference(
      * A list of [Issue]s that occurred handling this dependency.
      */
     val issues: List<Issue> = emptyList()
-) : Comparable<DependencyReference> {
-    /**
-     * Define an order on [DependencyReference] instances. Instances are ordered by their indices and fragment indices.
-     */
-    override fun compareTo(other: DependencyReference): Int =
-        if (pkg != other.pkg) {
-            pkg - other.pkg
-        } else {
-            fragment - other.fragment
-        }
-}
+)
```

Comment:
I'm not sure what the original motivation for this `compareTo()` implementation was... like, was the implementation more or less arbitrary, just to be able to use the class in sorted sets, or was there more to it, like and intention to sort by indices in some representation?

Maybe @oheger-bosch can chime in here to confirm that this change (by now, with the preceding changes) is uncritical?

Reply:
It's been a while... But as far as I remember, the purpose of the `compareTo()` implementation was to have a more deterministic order in dependency graphs. During tests, it popped up that the order in which items were added to a graph builder had an impact on the resulting graphs. They were isomorphic, but not identical. By defining an ordering, the effect could be reduced, but not fully eliminated.

Reply:
> "They were isomorphic, but not identical" 

Do you remember in what way this could be problematic? Was it non-deterministic without implementing `compareTo()` ?

Reply:
I am not 100% sure, but I think for a given order in which the graph builder was called, the results were the same.